test/test_optim.py

```diff
@@ -86,29 +99,51 @@ def _test_derived_optimizers(self, device, dtype, optim_info, flag):
                 state.append(optimizer.state)
                 updated_params.append(model.parameters())
 
+            assert_eq_kwargs = {}
+            if reduced_precision:
+                assert_eq_kwargs = {'atol': 1e-5, 'rtol': 1e-4}
```

Comment:
overriding precision is not great usually :/
Why do we need this? assertEqual should already adapt to low precision dtypes.

Reply:
the compounding of operators i think causes the default atol and rtol to not be big enough :/ This was added for fp16 default dtype test

Reply:
Ok, can we make these a top level concept and not bury it too deep?
Having one set of these reduced precision is fine, but we should make sure we don't get multiple